php-classes/Slate/CBL/Tasks/StudentTasksRequestHandler.php

```diff
@@ -162,75 +165,46 @@ public static function handleStudentTaskSubmissionRequest()
         ]);
     }
 
-    public static function setStudentTaskAttachments(StudentTask $Record, $data)
+    public static function setStudentTaskAttachments(StudentTask $Record, $requestData)
     {
-        if (isset($data['Attachments'])) {
-            $originalAttachments = $Record->Attachments;
-            $originalAttachmentIds = array_map(function($s) {
-                return $s->ID;
-            }, $originalAttachments);
 
-            $failed = [];
-            $attachmentIds = [];
-            $attachments = [];
-            $defaultAttachmentClass = AbstractTaskAttachment::$defaultClass;
+        $defaultAttachmentClass = AbstractTaskAttachment::class;

```

Comment:
@nbey `AbstractTaskAttachment` isn't actually declared as an `abstract` class (although it should be or should be named something else). If it really is an abstract class, we can't have an instance of it, and so it shouldn't be a default. An attachment without class=Link or class=GoogleDriveFile is invalid and should generate a validation error or exception, and the abstract class should be actually declared as such so that any attempt to instantiate one generates a proper error instead of just silent weirdness